http-server-netty/src/main/java/io/micronaut/http/server/netty/RoutingInBoundHandler.java

```diff
@@ -212,22 +239,169 @@ public void accept(ChannelHandlerContext ctx, io.netty.handler.codec.http.HttpRe
+        outboundAccess.attachment(mnRequest);
+        if (serverConfiguration.isOptimizedRouting() &&
+            request.decoderResult().isSuccess() &&
+            // origin needs to be checked by CorsFilter
+            !request.headers().contains(HttpHeaderNames.ORIGIN) &&
+            body instanceof ImmediateByteBody) {
+
+            PreparedMatchResult preparedMatchResult = routeExecutor.getRouter().findPreparedMatchResult(mnRequest);
+            if (preparedMatchResult != null) {
+                ExecutionLeaf<PreparedHandler> handler = preparedMatchResult.getHandler(PREPARED_HANDLER_KEY);
+                if (handler == null) {
+                    handler = shortCircuitHandler(preparedMatchResult.getRule(), preparedMatchResult.getRouteInfo());
+                    preparedMatchResult.setHandler(PREPARED_HANDLER_KEY, handler);
+                }
+                if (handler instanceof ExecutionLeaf.Route<PreparedHandler> route) {
+                    route.routeMatch().accept(mnRequest, outboundAccess);
+                    return;
+                }
+            }
+        }
         if (ctx.pipeline().get(ChannelPipelineCustomizer.HANDLER_ACCESS_LOGGER) != null) {
             // Micronaut Session needs this to extract values from the Micronaut Http Request for logging
             AttributeKey<NettyHttpRequest> KEY = AttributeKey.valueOf(NettyHttpRequest.class.getSimpleName());
             ctx.channel().attr(KEY).set(mnRequest);
         }
-        outboundAccess.attachment(mnRequest);
         try (PropagatedContext.Scope ignore = PropagatedContext.getOrEmpty().plus(new ServerHttpRequestContext(mnRequest)).propagate()) {
             new NettyRequestLifecycle(this, outboundAccess).handleNormal(mnRequest);
         }
     }
 
+    @Nullable
+    private static MatchRule.ContentType findFixedContentType(MatchRule matchRule) {
+        if (matchRule instanceof MatchRule.ContentType ct) {
+            return ct;
+        } else if (matchRule instanceof MatchRule.And and) {
+            return and.rules().stream()
+                .map(RoutingInBoundHandler::findFixedContentType)
+                .filter(Objects::nonNull)
+                .findFirst().orElse(null);
+        } else {
+            return null;
+        }
+    }
+
+    private ExecutionLeaf<PreparedHandler> shortCircuitHandler(MatchRule rule, UriRouteInfo<?, ?> routeInfo) {
+        if (routeInfo.isWebSocketRoute()) {
+            return ExecutionLeaf.indeterminate();
+        }
+        List<GenericHttpFilter> fixedFilters = routeExecutor.getRouter().getFixedFilters().orElse(null);
+        if (fixedFilters == null) {
+            return ExecutionLeaf.indeterminate();
+        }
+        // CorsFilter is handled specially here. It's always present, so we can't bail, but it only does anything when the Origin header is set, which is checked in accept().
+        fixedFilters = fixedFilters.stream().filter(f -> !FilterRunner.isCorsFilter(f, CorsFilter.class)).toList();
```

Comment:
Do we still need this? Filters can now be filtered by request, eliminating the CORS filter.

Reply:
i will profile